nsq/message.py

```diff
@@ -51,13 +51,22 @@ class Message(event.EventedMixin):
     def __init__(self, id, body, timestamp, attempts):
         self._async_enabled = False
         self._has_responded = False
+
         self.id = id
         self.body = body
         self.timestamp = timestamp
         self.attempts = attempts
+        self.timed_out = False
 
         super(Message, self).__init__()
 
+    def is_alive(self):
```

Comment:
OK, but I don't think it needs to be implemented by relying on some parent to set a property on this object - the message could gain the knowledge of its timeout and then just perform some time logic here?